### ISSUE TITLE
Potential Truncation of Number of Open Files

### ISSUE THREAD

**Issue by dmknutsen** [nasa/FM#27]
The number of open files is treated as a uint32 by FSW, but is telemetered as a uint8 such that truncation may result if the number of open files allowed by OSAL grows.

**Comment by skliper** [comment]
Switched to enhancement since this is informational and only in the HK packet (not the get open files command response telemetry) and basically just means we only support reporting up to uint8 files in HK.  Note the OSAL_CONFIG_MAX_NUM_OPEN_FILES parameter which is used to set OS_MAX_NUM_OPEN_FILES limits within OSAL the max number of open files which is currently defaulted to 50.

First off the tlm packet does support uint32:
https://github.com/nasa/FM/blob/3a8d00a5d6c74f7993a65a1b09372bb57920dc4e/fsw/src/fm_msg.h#L398-L404

OSAL has a config param defaulted to 50 for the maximum number of open files:
https://github.com/nasa/osal/blob/acb88caaf6507ff4bcae6efe56af81adba88252d/default_config.cmake#L258-L261

Logic for the command all supports uint32:
https://github.com/nasa/FM/blob/3a8d00a5d6c74f7993a65a1b09372bb57920dc4e/fsw/src/fm_cmds.c#L582-L611

Really the issue is in the HK packet (not the command response telemetry):
https://github.com/nasa/FM/blob/3a8d00a5d6c74f7993a65a1b09372bb57920dc4e/fsw/src/fm_msg.h#L448
https://github.com/nasa/FM/blob/3a8d00a5d6c74f7993a65a1b09372bb57920dc4e/fsw/src/fm_app.c#L422

**Comment by dmknutsen** [comment]
Agreed - that is what I was trying to communicate.  Thanks for the clarification!
